app/src/main/java/com/infomaniak/mail/ui/main/thread/AttachmentAdapter.kt

```diff
@@ -41,16 +44,36 @@ class AttachmentAdapter(
         val attachment = attachments[position]
 
         fileName.text = attachment.name
-        // TODO: Find how to add the FileType prefix before the file' size.
-        fileDetails.text = /*item.mimeType + " - " + */FormatterFileSize.formatShortFileSize(context, attachment.size.toLong())
+        fileDetails.text = FormatterFileSize.formatShortFileSize(context, attachment.size.toLong())
         icon.load(attachment.getFileTypeFromExtension().icon)
-        root.setOnClickListener { onAttachmentClicked?.invoke(attachment) }
+
+        if (!shouldDisplayCloseButton) {
+            root.setOnClickListener { onAttachmentClicked?.invoke(attachment) }
+        } else {
+            closeButton.apply {
+                setOnClickListener {
+                    val index = attachments.indexOf(attachment)
+                    if (index != -1) {
```

Comment:
When can it be -1?

Reply:
When clicking on the the attachment in order to delete it, we remove the attachment from the list successfully which means that during the fade out animation, if we click again on the little cross the attachment is not in the list anymore yet the cross has been clicked. This results in a negative index which is why this check is for

Reply:
Can you put a comment in the code to that end please?